### ISSUE TITLE
About Annotation in Validation sets

### ISSUE THREAD

**Issue by carol007** [VisualComputingInstitute/TrackR-CNN#16]
I follow the DAVIS data format to change the ImageSets of KITTIMOTS like 
![image](https://user-images.githubusercontent.com/20907787/61703032-54610700-ad73-11e9-95f4-424603fe902b.png)

Then I find first frame 000000.png in video 0006 has no foreground label value. all values are zero. And this may lead to unsupevised problem that cannot go through the VOS semi-supervised task based on 1st frame.

Is it reasonable to modify your valid.txt with videos that 1st frame absolutely has foreground ? 

Or

I hesitated how to use this valid videos to test  other method.  deleting unavailable frames at the beginning or I will use all annotations  without foreground at the beginning,and this may be unfair for evaluation.

**Comment by pvoigtlaender** [comment]
Hi,

KITTI MOTS was not really meant for video object segmentation, but ofcourse you can do it. Then probably start at the first non-empty frame as you suggested.
But since it wasn't designed for that use, there is no clear standard way / rules how to do it, so just come up with something reasonable.

**Comment by carol007** [comment]
@pvoigtlaender 
Thank you! But I still feel confused about the original KITTIMOTS validation set. Why are these images disordered in 0002 sequences? If I go through a VOS task whether I should set 0002/000070.png as the first frame and 000228 000162 followed to satisfied a fair evaluation comparison introduced in your paper? Or I just set 000000 as init followed by 000001 000002, which is reasonable but may be different to evaluation as yours.
![image](https://user-images.githubusercontent.com/20907787/61763280-760bce00-ae07-11e9-91e6-f614944006db.png)

**Comment by pvoigtlaender** [comment]
Hi,

I'm not sure where the order you are showing comes from. Just go through them in order of the filename/frame index

**Comment by carol007** [comment]
It is in your TrackR-CNN/datasets/KITTI/segtrack/validation.txt...

Would you mind uploading your train.txt and valid.txt? It is really confusing.


![image](https://user-images.githubusercontent.com/20907787/61789099-66f84080-ae46-11e9-847a-7b026609c105.png)

**Comment by pvoigtlaender** [comment]
Oh right, I think this file was just used to train instance segmentation where the temporal order does not matter. The order in this file is just kind of random.
I don't have this kind of file in a reasonable order, but if you need it, you can just use sorted(glob.glob("*.png")) on the folder?

**Comment by carol007** [comment]
So, does it means that you summarize the evaluation index with just images in the validation.txtf not the whole sequence?

**Comment by pvoigtlaender** [comment]
No, the whole sequence is used. Just use all the sequences in the order specified by the filenames/frame indices.
Please ignore the train.txt and valid.txt which you found in our code. They were used for a different purpose and we just did not remove it before uploading the code.

**Comment by carol007** [comment]
Thank you sincerely! I get your point. BTW, how did you figure out first-empty frame problem when you use the whole sequence?

**Comment by pvoigtlaender** [comment]
When we do MOTS (not VOS), then the first frame is not special. We didn't do VOS with first-frame ground truth on this dataset. So just come up with something reasonable (e.g. find first frame which has a non-background non-ignore instance).
